src/sage/data_structures/stream.py

```diff
@@ -1011,32 +1009,17 @@ def __init__(self, approximate_order, true_order=False):
             sage: C = Stream_uninitialized(0)
             sage: TestSuite(C).run(skip="_test_pickling")
         """
-        self._target = None
         if approximate_order is None:
             raise ValueError("the valuation must be specified for undefined series")
         super().__init__(False, true_order)
+        self._iter = None
         self._approximate_order = approximate_order
 
-    def iterate_coefficients(self):
-        """
-        A generator for the coefficients of ``self``.
-
-        EXAMPLES::
-
-            sage: from sage.data_structures.stream import Stream_uninitialized
-            sage: from sage.data_structures.stream import Stream_exact
-            sage: z = Stream_exact([1], order=1)
-            sage: C = Stream_uninitialized(0)
-            sage: C._target
-            sage: C._target = z
-            sage: n = C.iterate_coefficients()
-            sage: [next(n) for _ in range(10)]
-            [0, 1, 0, 0, 0, 0, 0, 0, 0, 0]
-        """
-        n = self._approximate_order
-        while True:
-            yield self._target[n]
-            n += 1
+    def define(self, coeff_stream):
```

Comment:
This needs doctests.

Reply:
The whole thing is not working at all.  I only tried to recall what I tried a year ago, and whether anything changed with the recent `_approximate_order` updates.